[16442] Gateway & Dashboard 5.3.13 RNs.mdx (#1306)
TykTechnologies/tyk-docs: developer-support/release-notes/gateway.mdx

## 5.3 Release Notes

### 5.3.13 Release Notes

#### Release Date 08 May 2026

#### Release Highlights

> Tyk Gateway resolves a set of related issues affecting Gateway registration with the Dashboard at scale for deployments using an **unlimited node license**, where mass registrations or rolling upgrades could leave Gateways stuck in registration loops.

Reviewer: ```suggestion
This release resolves a set of related issues affecting Gateway registration with the Dashboard at scale for deployments using an **unlimited node license**, where mass registrations or rolling upgrades could leave Gateways stuck in registration loops.
```

> This patch also upgrades Tyk Gateway to Golang 1.24.

Reviewer: ```suggestion

```

| :---------------------------------------------------------- | :---------------------- | :---------------------- | :-------- | 
| [GoLang](https://go.dev/dl/)                               | 1.24       | 1.24       | [Go plugins](/api-management/plugins/golang) must be built using Go 1.24 | 
| [Redis](https://redis.io/download/)  | 6.2.x, 7.x  | 6.2.x, 7.x  | Used by Tyk Gateway | 
| [OpenAPI Specification](https://spec.openapis.org/oas/v3.0.3) | v3.0.x      | v3.0.x          | Supported by [Tyk OAS](/api-management/gateway-config-tyk-oas#tyk-vendor-extension-reference)|

#### Deprecations

There are no deprecations in this release.

#### Upgrade instructions

If you are upgrading to 5.3.13, please follow the detailed [upgrade instructions](#upgrading-tyk).

#### Downloads

- [Docker image to pull](https://hub.docker.com/r/tykio/tyk-gateway/tags?page=&page_size=&ordering=&name=v5.3.13)
  - ```bash
    docker pull tykio/tyk-gateway:v5.3.13
    ``` 

- Helm charts
  - [tyk-charts v5.2.0](/developer-support/release-notes/helm-chart#5-2-0-release-notes)

- [Source code tarball for OSS projects](https://github.com/TykTechnologies/tyk/releases)

> 

Reviewer: ```suggestion

<Note>
Tyk Helm Charts are configured to install the LTS version of Tyk Gateway. You will need to modify them to install v5.3.13.
</Note>

```

#### Changelog
<a id="Changelog-v5.3.13"></a>

##### Changed

<AccordionGroup>

<Accordion title='Updated Tyk Gateway to Golang 1.24'>
Updated the Tyk Gateway to use Golang 1.24, reducing exposure to potential security vulnerabilities in older versions.
</Accordion>

</AccordionGroup>

##### Fixed

<AccordionGroup>

<Accordion title='Resolve Gateway registration failures at scale on unlimited node licenses'>
We have resolved a set of related issues affecting Gateway registration with the Dashboard at scale for deployments using an **unlimited node license**. During mass registrations or rolling upgrades, a combination of lock contention, excessive Redis load, and incorrect handling of `409 Conflict` responses could leave Gateways stuck in registration loops without the credentials needed to serve traffic.

Gateway registration is now significantly more robust at scale: registration requests are no longer serialized across the fleet, Gateways recover cleanly from transient `409 Conflict` responses instead of looping, and the Redis load generated during registration storms is substantially reduced.

> A dedicated fix for **limited node license** deployments will be provided in an upcoming release. 

Reviewer: ```suggestion
```
